graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphQLService.java

```diff
@@ -0,0 +1,201 @@
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.QueryParams;
+import com.linecorp.armeria.internal.server.JacksonUtil;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionInput.Builder;
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.GraphQL;
+import graphql.GraphqlErrorException;
+import graphql.com.google.common.collect.ImmutableMap;
+
+final class DefaultGraphQLService extends AbstractHttpService implements GraphQLService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultGraphQLService.class);
+
+    private static final ObjectMapper OBJECT_MAPPER = JacksonUtil.newDefaultObjectMapper();
+
+    private static final TypeReference<Map<String, Object>> JSON_MAP =
+            new TypeReference<Map<String, Object>>() {};
+
+    private final GraphQL graphQL;
+
+    @Nullable
+    private final DataLoaderRegistry dataLoaderRegistry;
+
+    DefaultGraphQLService(GraphQL graphQL) {
+        this(graphQL, null);
+    }
+
+    DefaultGraphQLService(GraphQL graphQL, @Nullable DataLoaderRegistry dataLoaderRegistry) {
+        this.graphQL = graphQL;
+        this.dataLoaderRegistry = dataLoaderRegistry;
+    }
+
+    @Override
+    protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest request) {
+        final QueryParams queryString = QueryParams.fromQueryString(ctx.query());
+        final String query = queryString.get("query");
+        if (query == null) {
+            return HttpResponse.of(HttpStatus.BAD_REQUEST,
+                                   MediaType.PLAIN_TEXT,
+                                   "Query is required");
+        }
+        final Map<String, Object> variables = toVariableMap(queryString.get("variables"));
+        final String operationName = queryString.get("operationName");
+        return execute(executionInput(query, ctx, variables, operationName));
+    }
+
+    @Override
+    protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest request) {
+        final MediaType contentType = request.contentType();
+        if (contentType == null) {
+            return HttpResponse.of(HttpStatus.UNPROCESSABLE_ENTITY,
+                                   MediaType.PLAIN_TEXT,
+                                   "Could not process GraphQL request");
+        }
+        if (contentType.is(MediaType.JSON) || contentType.subtype().endsWith("+json")) {
+            return HttpResponse.from(request.aggregate().handleAsync((req, thrown) -> {
+                if (thrown != null) {
+                    logger.warn("{} Failed to aggregate a request:", ctx, thrown);
```

Comment:
I guess this will leave too many logs on the server-side.
So, how about just removing this line?